### PR TITLE
@orta => use new image_urls json field

### PR DESCRIPTION
**Pull request by 1aurabrown**
the api now supplies a dictionary (`image_urls`) of image versions and their corresponding urls. we should use that instead of `image_version` and`image_url`. should fix https://github.com/artsy/eigen/issues/149

### CODE REVIEW

**Comment by 1aurabrown** [comment]
there may be other models that could benefit from making this switch. i only dealt with fairs because the fair view was experiencing missing images.

File: Artsy/Classes/View Controllers/ARParallaxHeaderViewController.m [hunk 1]
```diff
@@ -122,10 +115,20 @@ - (void)downloadImages
...
+                [self.iconImageView constrainWidth:@"80"];
+                [self.iconImageView constrainHeight:@"80"];
+
+                [self.titleLabel constrainLeadingSpaceToView:self.iconImageView predicate:@"20"];
+
                 // Necessary, since the icons will sometimes be transparent GIFs
                 self.iconImageView.backgroundColor = [UIColor whiteColor];
+            } else {
+                [self.titleLabel alignLeadingEdgeWithView:self.view predicate:@"20"];
+                [self.iconImageView removeFromSuperview];
```
Comment (1aurabrown) [inline Artsy/Classes/View Controllers/ARParallaxHeaderViewController.m]: avoid a grey square if the image is missing for some reason

File: Artsy Tests/ARParallaxHeaderViewControllerTests.m [hunk 2]
```diff
@@ -5,62 +5,87 @@
...
+
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    NSString *wideImagePath = [bundle pathForResource:@"wide" ofType:@"jpg"];
+    UIImage *wideImage = [UIImage imageWithContentsOfFile:wideImagePath];
+    NSString *squareImagePath = [bundle pathForResource:@"square" ofType:@"png"];
+    UIImage *squareImage = [UIImage imageWithContentsOfFile:squareImagePath];
+
+    [[SDImageCache sharedImageCache] storeImage:wideImage forKey:@"http://static1.artsy.net/fairs/52617c6c8b3b81f094000013/9/wide.jpg" toDisk:NO];
+    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square.png" toDisk:NO];
+    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square2.png" toDisk:NO];
```
Comment (orta) [inline Artsy Tests/ARParallaxHeaderViewControllerTests.m]: nice

Comment (1aurabrown) [inline Artsy Tests/ARParallaxHeaderViewControllerTests.m]: I saw tht this failed. I can take a peek later as i'll be getting home earlier than usual. Sent using CloudMagicOn Fri, Feb 13, 2015 at 14:54,  artsy/eigen <reply@reply.github.com> wrote:            In Artsy Tests/ARParallaxHeaderViewControllerTests.m:
> +        @"end_at" : @"2014-03-09T22:00:00.000+00:00",
> +        @"location" : @{
> +            @"city" : @"Toronto",
> +            @"state" : @"ON"
> +        }
> +    }];
> +
> +    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
> +    NSString *wideImagePath = [bundle pathForResource:@"wide" ofType:@"jpg"];
> +    UIImage *wideImage = [UIImage imageWithContentsOfFile:wideImagePath];
> +    NSString *squareImagePath = [bundle pathForResource:@"square" ofType:@"png"];
> +    UIImage *squareImage = [UIImage imageWithContentsOfFile:squareImagePath];
> +
> +    [[SDImageCache sharedImageCache] storeImage:wideImage forKey:@"http://static1.artsy.net/fairs/52617c6c8b3b81f094000013/9/wide.jpg" toDisk:NO];
> +    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square.png" toDisk:NO];
> +    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square2.png" toDisk:NO];

nice

Comment (orta) [inline Artsy Tests/ARParallaxHeaderViewControllerTests.m]: I think the Search on may be legit, but I feel like the artwork one isn't

Comment (1aurabrown) [inline Artsy Tests/ARParallaxHeaderViewControllerTests.m]: Everything passed locally for me before i pushed, but i've experienced intermittent failures with the ones that failed. Not sure what would have affected search in this PR. Sent using CloudMagicOn Fri, Feb 13, 2015 at 14:57,  artsy/eigen <reply@reply.github.com> wrote:            In Artsy Tests/ARParallaxHeaderViewControllerTests.m:
> +        @"end_at" : @"2014-03-09T22:00:00.000+00:00",
> +        @"location" : @{
> +            @"city" : @"Toronto",
> +            @"state" : @"ON"
> +        }
> +    }];
> +
> +    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
> +    NSString *wideImagePath = [bundle pathForResource:@"wide" ofType:@"jpg"];
> +    UIImage *wideImage = [UIImage imageWithContentsOfFile:wideImagePath];
> +    NSString *squareImagePath = [bundle pathForResource:@"square" ofType:@"png"];
> +    UIImage *squareImage = [UIImage imageWithContentsOfFile:squareImagePath];
> +
> +    [[SDImageCache sharedImageCache] storeImage:wideImage forKey:@"http://static1.artsy.net/fairs/52617c6c8b3b81f094000013/9/wide.jpg" toDisk:NO];
> +    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square.png" toDisk:NO];
> +    [[SDImageCache sharedImageCache] storeImage:squareImage forKey:@"http://static1.artsy.net/profile_icons/530cc50c9c18dbab9a00005b/square2.png" toDisk:NO];

I think the Search on may be legit, but I feel like the artwork one isn't

**Pull request merged**